### DISCUSSION TITLE
Akkuschonung Bug?

### DISCUSSION THREAD

**Discussion by iReckyy** [wiggal/GEN24_Ladesteuerung#122] [Q&A] [answered]
Hi, wie es scheint, stellt mir die Akkuschonung nur noch den Ladewert für die niedrigste Einstellung ein.

Das gleiche Problem wie bei den Notstromgrenzen?

Einstellungen: {"80": "0.25", "85": "0.2", "90": "0.15", "95": "0.1", "99": "0.05"}.

Log:
.
.
.
***** BEGINN: 2024-12-28 13:46:02 *****
## HTTP-LADESTEUERUNG ##
>>>Parameteränderung durch WebUI-Settings: Ladesteuerung:['logging', 'laden', 'entladen', 'optimierung', 'notstrom']
aktuellePrognose: 3668
TagesPrognose - BattVollUm: 803 - 14
Grundlast_Summe für Tag: 140
aktuellePVProduktion/Watt: 4395
aktuelleEinspeisung/Watt: 24
aktuelleBatteriePower/Watt: -1667
GesamtverbrauchHaus/Watt: 2704
aktuelleBattKapazität/Watt: 634
Batteriestatus in Prozent: 93.8 %
LadewertGrund: Akkuschonung: Ladestand >= 80%, Ladewert = 0.25C
Bisheriger Ladewert/Watt: 2560
Neuer Ladewert/Watt: 2560

## ENTLADESTEUERUNG ##

Feste Entladegrenze in % : 100 %
VerbrauchsgrenzeEntladung: 0 W
Feste Entladegrenze Table: 2000 W
Batterieentladegrenze ALT: 2000 W
Batterieentladegrenze NEU: 2000 W

Änderungen kleiner Schreibgrenze!

In SQLite-Datei gespeichert!

DEBUG *************** Berechnung Abzugswert:
DEBUG ##Schleife## Stunden_sum: 0.233, Prognose: 3828, Pro_Ertrag_Tag: 893

DEBUG *************** Ladewertmittel LOOP: 0
DEBUG ########### Pro_Akt_fun: 784.23 REST_Akt_Minute: 0.233
DEBUG 2024-12-28 12:00:00 Pro_Akt_fun: 784.23 Prognose_gesamt: 784.23

DEBUG *************** Ladewertmittel LOOP: 1
DEBUG 2024-12-28 13:00:00 Pro_Akt_fun: 3828 Prognose_gesamt: 4612.23

DEBUG *************** Ladewertmittel LOOP: 2
DEBUG ########### Pro_Akt_fun: 2723.97 Akt_Minute: 0.767
DEBUG 2024-12-28 14:00:00 Pro_Akt_fun: 2723.97 Prognose_gesamt: 7336.2

DEBUG 12/28/24 13:46 Aktuelle Prognose - Reservierung: 3668, Batteriekapazität: 634, aktuellerLadewert: 5000

DEBUG <<<<<< Meldungen von Akkuschonung >>>>>>>
DEBUG Akkuschonung_Werte: {'99': '0.05', '95': '0.1', '90': '0.15', '85': '0.2', '80': '0.25'}
DEBUG AkkuschonungLadewert-alterLadewert: 0
DEBUG BattStatusProz_Grenze: 76.0
DEBUG AkkuschonungLadewert: 2560
DEBUG aktuellerLadewert: 5000

DEBUG BattVollUm: 14Uhr
DEBUG WRSchreibGrenze_nachUnten: 512.0W
DEBUG WRSchreibGrenze_nachOben: 512.0W
DEBUG <<<<<<<< ENTLADESTEUERUNG >>>>>>>>>>>>>
DEBUG MaxEntladung = entladesteurungsdata:10240
DEBUG VerbrauchsgrenzeEntladung aus Spalte 1: 0
DEBUG FesteEntladegrenze aus Spalte 2: 2000
DEBUG Batterieentladegrenze NEU: 2000%
DEBUG <<<<<<<< ENDE ENTLADESTEUERUNG >>>>>>>>>>>>>

DEBUG <<<<<<<< Eigenverbrauchs-Optimierung >>>>>>>>>>>>>
DEBUG ## Dauer_Nacht_Std: 0.23, Akku_Rest_Watt: 8488
DEBUG ## Eigen_Opt_genau: 36469, Eigen_Opt_Std_neu: 36400
DEBUG ## >>> Bei PrognoseMorgen < PrognoseGrenzeMorgen, keine Einspeisung während des Tages
DEBUG ## >>> PrognoseMorgen: 17.267, PrognoseGrenzeMorgen: 40
DEBUG ## >>> EigenverbOpt_steuern == 2, keine Einspeisung während des Tages
***** ENDE: 2024-12-28 13:46:02 *****

***** BEGINN: 2024-12-28 13:51:02 *****
## HTTP-LADESTEUERUNG ##
>>>Parameteränderung durch WebUI-Settings: Ladesteuerung:['logging', 'laden', 'entladen', 'optimierung', 'notstrom']
aktuellePrognose: 3676
TagesPrognose - BattVollUm: 516 - 14
Grundlast_Summe für Tag: 90
aktuellePVProduktion/Watt: 4322
aktuelleEinspeisung/Watt: 759
aktuelleBatteriePower/Watt: -2531
GesamtverbrauchHaus/Watt: 1032
aktuelleBattKapazität/Watt: 409
Batteriestatus in Prozent: 96.0 %
LadewertGrund: Akkuschonung: Ladestand >= 80%, Ladewert = 0.25C
Bisheriger Ladewert/Watt: 2560
Neuer Ladewert/Watt: 2560

## ENTLADESTEUERUNG ##

Feste Entladegrenze in % : 100 %
VerbrauchsgrenzeEntladung: 0 W
Feste Entladegrenze Table: 2000 W
Batterieentladegrenze ALT: 2000 W
Batterieentladegrenze NEU: 2000 W

Änderungen kleiner Schreibgrenze!

In SQLite-Datei gespeichert!

DEBUG *************** Berechnung Abzugswert:
DEBUG ##Schleife## Stunden_sum: 0.15, Prognose: 3828, Pro_Ertrag_Tag: 574

DEBUG *************** Ladewertmittel LOOP: 0
DEBUG ########### Pro_Akt_fun: 504.15 REST_Akt_Minute: 0.15
DEBUG 2024-12-28 12:00:00 Pro_Akt_fun: 504.15 Prognose_gesamt: 504.15

DEBUG *************** Ladewertmittel LOOP: 1
DEBUG 2024-12-28 13:00:00 Pro_Akt_fun: 3828 Prognose_gesamt: 4332.15

DEBUG *************** Ladewertmittel LOOP: 2
DEBUG ########### Pro_Akt_fun: 3020.05 Akt_Minute: 0.85
DEBUG 2024-12-28 14:00:00 Pro_Akt_fun: 3020.05 Prognose_gesamt: 7352.2

DEBUG 12/28/24 13:51 Aktuelle Prognose - Reservierung: 3676, Batteriekapazität: 409, aktuellerLadewert: 5000

DEBUG <<<<<< Meldungen von Akkuschonung >>>>>>>
DEBUG Akkuschonung_Werte: {'99': '0.05', '95': '0.1', '90': '0.15', '85': '0.2', '80': '0.25'}
DEBUG AkkuschonungLadewert-alterLadewert: 0
DEBUG BattStatusProz_Grenze: 76.0
DEBUG AkkuschonungLadewert: 2560
DEBUG aktuellerLadewert: 5000

DEBUG BattVollUm: 14Uhr
DEBUG WRSchreibGrenze_nachUnten: 512.0W
DEBUG WRSchreibGrenze_nachOben: 512.0W
DEBUG <<<<<<<< ENTLADESTEUERUNG >>>>>>>>>>>>>
DEBUG MaxEntladung = entladesteurungsdata:10240
DEBUG VerbrauchsgrenzeEntladung aus Spalte 1: 0
DEBUG FesteEntladegrenze aus Spalte 2: 2000
DEBUG Batterieentladegrenze NEU: 2000%
DEBUG <<<<<<<< ENDE ENTLADESTEUERUNG >>>>>>>>>>>>>

DEBUG <<<<<<<< Eigenverbrauchs-Optimierung >>>>>>>>>>>>>
DEBUG ## Dauer_Nacht_Std: 0.15, Akku_Rest_Watt: 8746
DEBUG ## Eigen_Opt_genau: 58547, Eigen_Opt_Std_neu: 58500
DEBUG ## >>> Bei PrognoseMorgen < PrognoseGrenzeMorgen, keine Einspeisung während des Tages
DEBUG ## >>> PrognoseMorgen: 17.267, PrognoseGrenzeMorgen: 40
DEBUG ## >>> EigenverbOpt_steuern == 2, keine Einspeisung während des Tages
***** ENDE: 2024-12-28 13:51:02 *****

**Comment by wiggal** [comment, marked as answer]
Ups, da ist beim fixen ein neuer Fehler in der Akkuschonung reingeflutscht, sorry.
Habe es im `main` gefixt

**Reply by iReckyy** [reply]
Akkuschonung-Bug ist behoben.
